pkg/controller/build/reconciler.go

```diff
@@ -101,7 +100,8 @@ func (b *buildReconciler) updateMachineOSConfig(ctx context.Context, old, cur *m
 	}
 
 	// Whenever the build inputs have changed, create a new MachineOSBuild.
-	if !equality.Semantic.DeepEqual(old.Spec.BuildInputs, cur.Spec.BuildInputs) {
+	// QOCL: ok to compare the spec.MachineConfigPool by comparing the whole spec here? Inputs didnt include the pool before.
```

Comment:
It's OK in my opinion. What I don't know is whether the MCP reference is immutable. In my opinion, it should be.